contracts/AcceleratingDistributor.sol

```diff
@@ -156,30 +154,36 @@ contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
         // payouts could eat into user balances. maxMultiplier is constrained to be at least 1e18 to enforce a minimum
         // 1x multiplier and avoid potential underflows.
         require(stakedToken != address(rewardToken), "Staked token is reward token");
+        require(baseEmissionRate <= 1e24, "baseEmissionRate too large"); // 1 million tokens per second.
         require(maxMultiplier <= 1e24, "maxMultiplier too large"); // 1_000_000x multiplier.
         require(maxMultiplier >= 1e18, "maxMultiplier less than 1e18");
         require(secondsToMaxMultiplier > 0, "secondsToMaxMultiplier is 0");
-        require(baseEmissionRate <= 1e24, "baseEmissionRate too large"); // 1 million tokens per second.
 
         StakingToken storage stakingToken = stakingTokens[stakedToken];
+        require(stakingToken.lastUpdateTime == 0, "Staking token already added");
 
-        // If this token is already initialized, make sure we update the rewards before modifying any params.
-        if (stakingToken.lastUpdateTime != 0) _updateReward(stakedToken, address(0));
-
-        stakingToken.enabled = enabled;
         stakingToken.baseEmissionRate = baseEmissionRate;
         stakingToken.maxMultiplier = maxMultiplier;
         stakingToken.secondsToMaxMultiplier = secondsToMaxMultiplier;
-        stakingToken.lastUpdateTime = getCurrentTime();
 
-        emit TokenConfiguredForStaking(
-            stakedToken,
-            enabled,
-            baseEmissionRate,
-            maxMultiplier,
-            secondsToMaxMultiplier,
-            stakingToken.lastUpdateTime
-        );
+        emit TokenAddedForStaking(stakedToken, baseEmissionRate, maxMultiplier, secondsToMaxMultiplier);
+    }
+
+    /**
+     * @notice Change the configuration of a staking token. Emits a StakingTokenUpdated event on success.
+     * @param stakedToken The address of the token that can be staked.
+     * @param enabled Whether the token is enabled for staking.
+     */
+    function configureStakingToken(address stakedToken, bool enabled) external onlyOwner {
+        require(stakedToken != address(0), "Invalid staking token");
+
+        StakingToken storage stakingToken = stakingTokens[stakedToken];
+        require(stakingToken.maxMultiplier >= 1e18, "Staking token not added");
+
+        _updateReward(stakedToken, address(0));
```

Comment:
nb. This is a departure from the existing logic in configureStakingToken(). The main driver for changing it was to prevent calling addStakingToken() again after the token has been enabled for the first time. That could also be done by checking other members - i.e. baseEmissionRate, maxMultiplier, or secondsToMaxMultiplier).